src/main/java/hillel/doctorRest/clinic/review/ReviewService.java

```diff
@@ -0,0 +1,83 @@
+package hillel.doctorRest.clinic.review;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import lombok.val;
+
+@Service
+@AllArgsConstructor
+
+public class ReviewService {
+
+    private final ReviewRepository reviewRepository;
+
+    public Review createReview(Review review) {
+        return reviewRepository.save(review);
+    }
+
+    public void saveReview(Review review) {
+        reviewRepository.save(review);
+    }
+
+    public Optional<Review> findById(Integer id) {
+        return reviewRepository.findById(id);
+    }
+
+    public List<Review> findAll() {
+        return reviewRepository.findAll();
+    }
+
+    public OptionalDouble averageService() {
+        return findAll().stream().map(review -> review.getService())
+                .flatMap(Optional::stream).mapToDouble(value -> value).average();
+    }
+
+    public OptionalDouble averageEquipment() {
+        return findAll().stream().map(review -> review.getEquipment())
+                .flatMap(Optional::stream).mapToDouble(value -> value).average();
+    }
+
+    public OptionalDouble averageQualificationSpecialist() {
+        return findAll().stream().map(review -> review.getQualificationSpecialist()
+        ).flatMap(Optional::stream).mapToDouble(value -> value).average();
+    }
+
+    public OptionalDouble averageEffectivenessOfTreatment() {
+        return findAll().stream().map(review -> review.getEffectivenessOfTreatment()
+        ).flatMap(Optional::stream).mapToDouble(value -> value).average();
+    }
+
+    public OptionalDouble averageRatingOverall() {
+        return findAll().stream().map(review -> review.getRatingOverall())
+                .flatMap(Optional::stream).mapToDouble(value -> value).average();
+    }
+
+    public Map<String, OptionalDouble> mapCriterionToAverage() {
+        val map = new HashMap<String, OptionalDouble>();
+        map.put("averageService", averageService());
+        map.put("averageEquipment", averageEquipment());
+        map.put("averageQualificationSpecialist", averageQualificationSpecialist());
+        map.put("averageEffectivenessOfTreatment", averageEffectivenessOfTreatment());
+        map.put("averageRatingOverall", averageRatingOverall());
+        return map;
```

Comment:
It is better to use a Class with statically typed fields, instead of this map with dynamic keys.